### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by bizip**
**What does this PR do?**

Add the following validations:

- For the User model:

- Name must not be blank.

- PostsCounter must be an integer greater than or equal to zero.

- For the Post model:

- Title must not be blank.

- Title must not exceed 250 characters.

- CommentsCounter must be an integer greater than or equal to zero.

- LikesCounter must be an integer greater than or equal to zero.

- Add unit specs for all of your models' methods and validations.

**Description of Task to be completed**

All tasks are completed 

**How should this be manually tested?**

1. Clone this repository and switch to `test-validattions`

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,9 +1,11 @@
-class User < ActiveRecord::Base
+class User < ApplicationRecord
+  validates :name, presence: true
+
```
Comment (phelian23) [inline app/models/user.rb]: - [ ] As requested in the project requirements, kindly add validation for the post counter in the user model

File: app/models/post.rb [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
+  validates :title, presence: true, length: { maximum: 250 }
 
-  def five_recent_comment
-    posts.limit(5).order(comments(:desc))
-  end
-
-  after_save :update_post_counter
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id', counter_cache: true
+  has_many :likes
+  has_many :comments
```
Comment (phelian23) [inline app/models/post.rb]: - [ ] As requested in the project requirements, kindly add a validation for comments counter and likes counter in the post model

**Review by phelian23 — CHANGES_REQUESTED** [review]
Hi @bizip,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
Great job setting up the correct linters ✔️
Professional documentation ✔️
Correct workflow ✔️
Meaningful commit messages ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
Hi @bizip, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**